portals/portals.c

```diff
@@ -91,6 +92,16 @@ static void remove_swptl_dev(struct linked_list **start, struct swptl_dev *dev)
 	}
 }
 
+static void free_dev_list()
+{
+	struct linked_list *current = dev_list;
+	while (current != NULL) {
+		struct linked_list *n = current->next;

```

Comment:
Shouldn't we call `swptl_dev_del` in this function?
You can check if there is a memory leak when doing only PtlFini, and not the other cleanup functions (in particular not PtlNIFini)